code/game/objects/items/storage/book.dm

```diff
@@ -207,20 +207,21 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "burning",
 
 	else if(istype(A, /obj/item/soulstone) && !iscultist(user))
 		var/obj/item/soulstone/SS = A
-		to_chat(user, "<span class='notice'>You begin to exorcise [SS].</span>")
-		playsound(src,'sound/hallucinations/veryfar_noise.ogg',40,1)
-		if(do_after(user, 40, target = SS))
-			playsound(src,'sound/effects/pray_chaplain.ogg',60,1)
-			SS.usability = TRUE
-			SS.purified = TRUE
-			SS.icon_state = "purified_soulstone"
-			for(var/mob/M in SS.contents)
-				if(M.mind)
-					SS.icon_state = "purified_soulstone2"
-			for(var/mob/living/simple_animal/shade/EX in SS)
-				EX.icon_state = "ghost1"
-				EX.name = "Purified [EX.name]"				
-			user.visible_message("<span class='notice'>[user] has purified the [SS]!</span>")
+		if(!SS.purified)
```

Comment:
i suggest...
`if(SS.purified)`
and then returning, it will stop the code from going deeper than it needs to

Reply:
This would run into the same issue stated above, though the two need not be mutually exclusive.

Reply:
I'd recommend both actually. This one to preemptively halt and the other for sanity.

There's nothing wrong with duplicate checks since they're not REALLY duplicates in terms of purpose.